eng/pipelines/templates/stages/archetype-sdk-client-patch.yml

```diff
@@ -109,6 +108,17 @@ stages:
             pip install markdown2 BeautifulSoup4
           displayName: 'pip install markdown2 and BeautifulSoup4'
 
+        # Save the Package Properties
+        # ServiceDirectories variable is a plain, comma separated list of ServiceDirectories that is
+        # output by the generate_from_source_pom script above.
+        - pwsh: |
+            $ServiceDirs = '$(ServiceDirectories)'
+            foreach($ServiceDir in $ServiceDirs.Split(',')) {
+              $(Build.SourcesDirectory)/eng/common/scripts/Save-Package-Properties.ps1 -ServiceDirectory $ServiceDir -OutDirectory $(Build.ArtifactStagingDirectory)/PackageInfo
```

Comment:
Won't this save the info for all the projects in a service directory? Don't we only want particular artifacts?

Reply:
Yes, and that's unfortunately because of the way it the Save-Package-Properties.ps1 was written. It'll always create them for everything within the service directory. The mitigation here is that the only thing that uses them is the docs.ms publishing which takes the list [PackageInfoLocations](https://github.com/Azure/azure-sdk-for-java/pull/30279/files#diff-20f9d9754d8d3c5cb33a9557c11a406a93b41bd3b15a54e268fb0c8fc1374161R262-R265) that is explicitly computed for each artifact. The extras don't matter. This isn't any different from batch release, where you're selecting what libraries to release via the checkboxes, and the Save-Package-Properties.ps1 is still going to create a PackageInfo for each Artifact in the service directory. Lastly, this literally takes less than a second to run even for a ServiceDirectory with as many libraries as Spring, I'm not worried about the time.

Reply:
OK yeah I'm good with this as long as we don't make decisions on the existence of these and we still filter down based on the artifact list.